library/network/src/lib/y2firewall/firewalld/zone.rb

```diff
@@ -63,12 +66,27 @@ class Zone
       # @param name [String] zone name
       def initialize(name: nil)
         @name = name || api.default_zone
+        @modified = []
       end
 
       def self.known_zones
         KNOWN_ZONES
       end
 
+      # Setter method for enabling masquerading.
+      #
+      # @param enabled [Boolean] true for enable; false for disable
+      # @return [Boolean] whether it is enabled or not
+      def masquerade=(enable)
+        @modified << :masquerade unless @modified.include?(:masquerade)
```

Comment:
same or very similar code is at multiple places, so it might deserve own method

Reply:
Ok, added the cache option directly into the `Relations` module. It can be enable directly when the relations are declared just passing the argument `use_cache: true`

```ruby
 class Zone
   extend Relations
   has_many :interfaces, use_cache: true

   def initialize
     @modified = []
   end
 end
```